test/distributed/org/apache/cassandra/distributed/upgrade/MixedModeFrom3ReplicationTest.java

```diff
@@ -18,21 +18,65 @@
 
 package org.apache.cassandra.distributed.upgrade;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.junit.Test;
 
-import org.apache.cassandra.distributed.shared.Versions;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
 
-public class MixedModeFrom3ReplicationTest extends MixedModeReplicationTestBase
-{
-    @Test
-    public void testSimpleStrategy30to3X() throws Throwable
-    {
-        testSimpleStrategy(v30, v3X);
-    }
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
 
+public class MixedModeFrom3ReplicationTest extends UpgradeTestBase
```

Comment:
`MixedModeReplicationTestBase` only was extended by this class, so was simpler to just inline all the logic here...